### PR TITLE
Add abiotic documentation

### PR DESCRIPTION
**Pull request by vgro**
This PR adds the theory for the abiotic components of the Virtual Ecosystem. So far, I have added the overview, microclimate, and a draft of hydrology. I would appreciate feedback on structure and content before I put a lot of effort in adding references and repeating the exercise for the hydrology.

You can read the docs here
https://virtual-ecosystem.readthedocs.io/en/add_abiotic_documentation/virtual_ecosystem/abiotic_theory.html
https://virtual-ecosystem.readthedocs.io/en/add_abiotic_documentation/virtual_ecosystem/microclimate_theory.html
https://virtual-ecosystem.readthedocs.io/en/add_abiotic_documentation/virtual_ecosystem/hydrology_theory.html

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Key checklist

- [x] Make sure you've run the `pre-commit` checks: `$ pre-commit run -a`
- [ ] All tests pass: `$ poetry run pytest`

## Further checks

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works
- [x] Relevant documentation reviewed and updated

### CODE REVIEW

File: docs/source/virtual_ecosystem/microclimate_theory.md [hunk 1]
```diff
@@ -0,0 +1,288 @@
+# Microclimate
+
+Microclimates are defined as the local climate conditions that organisms and ecosystems
+are exposed to. In terrestrial ecosystems, microclimates often deviate strongly from the
+climate representative of a large geographic region, the macroclimate
+{cite}`kemppinen_microclimate_2024`.
+For example, the temperature directly above a rainforest canopy might be modulated the
+due to small scale variations in topography and aspect. The temperature above the canopy
```
Comment (jacobcook1995) [inline docs/source/virtual_ecosystem/microclimate_theory.md:8]: "modulated due to"

File: docs/source/virtual_ecosystem/microclimate_theory.md [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+the surface and vegetation.
+* **Energy balance**: The energy balance describes the equilibrium of absorbed and released
+energy at a surface, for example the soil surface or the canopy. This balance is closely
+coupled to the radiation balance through net radiation, which is partitioned into
+turbulent fluxes (latent, sensible, and ground heat flux), used for photosynthesis, and
+changes in heat storage.
+* **Water balance**: The water balance refers to the equilibrium of absorbed and released
+water by different (here abiotic) ecosystem components. This balance is linked to the
+energy balance via evapotranspiration and latent heat flux from the soil surface.
+The hydrology on catchment scale is described in more detail [here](./microclimate_theory.md).
```
Comment (jacobcook1995) [inline docs/source/virtual_ecosystem/microclimate_theory.md:102]: Guess the link here should be to `./hydrology_theory.md`?

File: docs/source/virtual_ecosystem/microclimate_theory.md [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+* **Carbon balance**: The carbon balance is linked to the radiation, energy and water
+balance by net primary productivity: the conversion of light, atmospheric carbon, water,
+(and nutrients) into biomass minus respiration. The carbon cycle continues as plant biomass
+is either eaten by herbivores or falls to the ground. If not respired by animals or
+plants, carbon enters the soil where it is and eventually recycled to the atmosphere.
```
Comment (jacobcook1995) [inline docs/source/virtual_ecosystem/microclimate_theory.md:107]: "where it is decomposed" (maybe)?

File: docs/source/virtual_ecosystem/microclimate_theory.md [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+Pressure decreases with height and influences weather patterns; changes in atmospheric
+pressure can indicate upcoming weather changes, such as storms.
+
+### Atmospheric $\ce{CO_{2}}$
+
+Although atmospheric $\ce{CO_{2}}$ is not a microclimatic variable, it is closely tied
+in the balance of energy through its critical role in photosynthesis. It should therefore
+be considered when studying the dynamics on ecosystems.
+
+## Links to biotic components
```
Comment (jacobcook1995) [inline docs/source/virtual_ecosystem/microclimate_theory.md:183]: I feel like this section could maybe do with a (small) introduction, as it feels a bit of an abrupt change after the "key microclimate variables" section

**Review by jacobcook1995 — APPROVED** [review]
I think this is really good! Only had a few minor comments

**Review by davidorme — APPROVED** [review]
LGTM too. 

Both the hydrology and microclimate pages present lists of factors, variables, biotic links etc as well as the key equations. The pages are quite long  and I think it would be useful if there was short bit at the top that gives internal page links to (i) explain what is included in the page and (ii) help navigate it.  Something as simple as:

```
This page provides an overview of the [main equations](#link) underlying the 
hydrology along with the [key factors](#link) influencing these processes, the 
required [input and state variables](#link) used to estimate the processes and 
the [interactions with biotic process](#link).
```

I think it might be worth seeing if we can make the trees in the sketch image more visible? They've got a bit lost in the digitisation process.

**Pull request merged**